Microsoft.Research/CCDoc/BooleanExpressionHelper.cs

```diff
@@ -341,5 +341,23 @@ internal class ExpressionSimplifier : CodeRewriter {
       }
       return base.Rewrite(conditional);
     }
+
+      public override IExpression Rewrite(IGreaterThan greaterThan)
+      {
+          // Catch use of the x > null idiom used by Roslyn as generated
+          // code for x != null statements.
+          if (ExpressionHelper.IsNullLiteral(greaterThan.RightOperand))
+          {
+              var c = new NotEquality
```

Comment:
Just a suggestion, I would suggest to remove local or use more clear name, like `result`, `nonEquality` or something else.